pykafka/rdkafka/simple_consumer.py

```diff
@@ -0,0 +1,239 @@
+from contextlib import contextmanager
+import logging
+
+from pykafka.exceptions import RdKafkaStoppedException, ConsumerStoppedException
+from pykafka.simpleconsumer import SimpleConsumer, OffsetType
+from pykafka.utils.compat import get_bytes
+from . import _rd_kafka
+
+
+log = logging.getLogger(__name__)
+
+
+class RdKafkaSimpleConsumer(SimpleConsumer):
+    """A pykafka.SimpleConsumer with librdkafka-based fetchers
+
+    This aims to conform to the SimpleConsumer interface as closely as
+    possible.  There are some notable differences:
+
+    1. rotating over partitions: while message ordering within partitions is
+       conserved (of course!), the order in which partitions are visited will
+       deviate.  In particular, here we may emit more than one message from
+       the same partition before visiting another
+    2. ignores num_consumer_fetchers: librdkafka will typically spawn at least
+       as many threads as there are kafka cluster nodes
+
+    For an overview of how configuration keys are mapped to librdkafka's, see
+    _mk_rdkafka_config_lists.
+    """
+    def __init__(self,
+                 topic,
+                 cluster,
+                 consumer_group=None,
+                 partitions=None,
+                 fetch_message_max_bytes=1024 * 1024,
+                 num_consumer_fetchers=1,
```

Comment:
Argh, can't remove it because it's passed in when instantiating the internal consumer in a `BalancedConsumer`. A possible option to make this signature more explicit is to remove `num_consumer_fetchers` and add an unused `**kwargs` to the end of the args list, but that seems far worse for comprehensibility. The only other alternatives I'm aware of involve dynamically building the kwargs list that's used on the internal consumer. Maybe that's actually better, but not a pressing issue. For now, the docstring handles it just fine.